crates/ra_ide/src/inlay_hints.rs

```diff
@@ -235,8 +235,19 @@ fn should_show_param_hint(
     param_name: &str,
     argument: &ast::Expr,
 ) -> bool {
-    let argument_string = argument.syntax().to_string();
-    if param_name.is_empty() || argument_string.ends_with(param_name) {
+    let argument_string = {
```

Comment:
Indeed, this is better expressed as `if let ast::RefExpr(argument) = argument`.

Probably better expressed as a helper function:

```
fn remove_reference(exr: ast::Expr) -> ast::Expr
```

Reply:
So, I tried to make a function that strips the Ref part, but I ended up with a token which I didn't know how to convert back to an ast::Expr. Instead I made a `fn is_argument_similar_to_param(argument, param_name) -> bool`. It makes the intent pretty explicit, but it's less reusable.